oak_crypto/src/encryptor.rs

```diff
@@ -210,3 +219,70 @@ impl ServerEncryptor {
         Ok(response)
     }
 }
+
+/// Encryptor object for decrypting client requests that are received by the server and encrypting
+/// server responses that will be sent back to the client. Each Encryptor object corresponds to a
+/// single crypto session between the client and the server.
+/// Encryptor state is initialized after receiving an initial request message containing client's
+/// encapsulated public key.
+///
+/// Sequence numbers for requests and responses are incremented separately, meaning that there could
+/// be multiple responses per request and multiple requests per response.
+// TODO(#4311): Merge `AsyncServerEncryptor` and `ServerEncryptor` once there is `async` support in
+// the Restricted Kernel.
+pub struct AsyncServerEncryptor {
+    recipient_context_generator: Arc<dyn AsyncRecipientContextGenerator>,
+    inner: Option<ServerEncryptor>,
+}
+
+impl AsyncServerEncryptor {
```

Comment:
Initially I thought this would be an async version of ServerEncryptor, instead it looks like the semantics is different, in particular AsyncServerEncryptor is more stateful. Is this intended?

Reply:
It's the only way to impelemt this. Becaise `AsyncServerEncryptor` is created without an initialized context, and it's initialized when `decrypt` is called the first time.